runtime/compiler/control/CompilationThread.cpp

```diff
@@ -10596,6 +10608,12 @@ TR::CompilationInfo::compilationEnd(J9VMThread * vmThread, TR::IlGeneratorMethod
 #if defined(J9VM_OPT_JITSERVER)
       if (entry && isJITServerMode) // failure at the JITServer
          {
+         // If no error code is set but the compilation failed,
+         // it means the diagnostic JitDump compilation crashed.
```

Comment:
" If no error code is set but the compilation failed, it means the diagnostic JitDump compilation crashed."
Don't we have some more reliable way of detecting this situation? At the very least we would need an assert that this is indeed a diagnostic compilation.

Reply:
Yeah, I'm not entirely sure about this part. From what I understand, this case shouldn't be a concern for a non-diagnostic compilation, because a regular compilation failure (e.g. exception thrown) will set the error code correctly, while a crash during a regular compilation will trigger a crash signal handler and this part of the code will never be reached. It's only for JitDump recompilations that a crashing compilation will reach this point, because its signal handler doesn't do anything.